In [6]:
import joblib
from sklearn.dummy import DummyClassifier
from sklearn.metrics import accuracy_score, f1_score
x_train = joblib.load('../data/x_train.pkl')
x_test = joblib.load('../data/x_test.pkl')
y_train = joblib.load('../data/y_train.pkl')
y_test = joblib.load('../data/y_test.pkl')

In [7]:
#baseline model
dummy = DummyClassifier(strategy="most_frequent", random_state=42)
dummy.fit(x_train, y_train)
y_pred = dummy.predict(x_test)

In [9]:
#Evaluation
accuracy = accuracy_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred, pos_label=1)
print(f"Accuracy: {accuracy:.4f}")
print(f"F1 Score: {f1:.4f}")

Accuracy: 0.7725
F1 Score: 0.0000


The baseline model shows an accuracy score of 77% which though sounding good is actually misleading, as there is class imbalance and the baseline mo0del exactly ignores the minority class. Accuracy ia a wrong metric here as it does not catch returns, recall can be a suitable metric though with 0 %, which can  be used as baseline to judge  other models.